state-chain/pallets/cf-elections/src/electoral_systems/mocks.rs

```diff
@@ -79,36 +83,48 @@ where
 	// Useful for testing check_consensus since we already have an election.
 	pub fn build_with_initial_election(self) -> TestContext<ES> {
 		let setup = self.clone();
-		let mut electoral_access = MockAccess::<ES>::new(
-			self.unsynchronised_state,
-			self.unsynchronised_settings,
-			self.electoral_settings,
-		);
+
+		// We need to clear the storage at every build so if there are multiple test contexts used
+		// within a single test they do not conflict.
+		MockStorageAccess::clear_storage();
```

Comment:
Do we really need this? Storage is not persisted between tests.

Reply:
Maybe we encapsulate both this and the intitialisation stuff below in a method MockStorageAccess::new(..)?

Reply:
> Do we really need this? Storage is not persisted between tests.

thread local storage is persisted between tests

> Maybe we encapsulate both this and the intitialisation stuff below in a method MockStorageAccess::new(..)?

This function is kind of doing that already, what are you suggesting exactly?